crates/ratchet-core/src/dtype/mod.rs

```diff
@@ -61,6 +48,13 @@ impl DType {
             _ => false,
         }
     }
+
+    pub fn is_float(self) -> bool {
+        match self {
+            DType::F32 | DType::F16 | DType::BF16 => true,
+            _ => false,
+        }
```

Comment:
```suggestion
        matches!(DType::F32 | DType::F16 | DType::BF16)
```

`is_float` could also be an associated const on `DType` if you prefer.